app/src/tests/components/contents/download/DownloadButton.test.tsx

```diff
@@ -0,0 +1,81 @@
+const mockDownload = jest.fn();
+jest.mock("../../../../lib/download", () => ({
+        download: async (...args: any[]) => mockDownload(...args)
+    })
+);
+
+import { render, screen, waitFor } from "@testing-library/react";
+import DownloadButton from "../../../../app/components/contents/download/DownloadButton";
+import {LoginState} from "../../../../types";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import {mockLoginState} from "../../../mocks";
+import userEvent from "@testing-library/user-event";
+
+describe("DownloadButton", () => {
+    const file = {
+        path: "test.txt",
+        size: 1024,
+        hash: "fakeHash"
+    };
+
+    const getStore = (props: Partial<LoginState> = {isAuthenticated: true}) => {
```

Comment:
as above dont think we neeed to test if user is authenticated or not

Reply:
This is because we might be running an instance with auth switched off.

Reply:
Actually, we could just let the getAuthHeader method return nothing if there's no current user, that might be simpler as the clients of the function wouldn't need to know, but might be confusing since the header returned might be empty,

Reply:
ohh yeah that auth instance off scenario

Reply:
It can be downloaded if there's no header and auth is switched off on the server, but you'll get a 401 if auth is switched on and no valid header is provided. But in terms of front end testing, we can just check that we provide the token if we have one.